contracts/strategy/TradeFintechStrategy.sol

```diff
@@ -0,0 +1,153 @@
+        uint256 _depositLimit
+    ) BaseStrategy(_asset, _name, _tokenizedStrategyAddress) {
+        if (_depositPeriodEnds == 0 || _lockPeriodEnds == 0 || _depositLimit == 0) {
+            revert ZeroValue();
+        }
+        if (_depositPeriodEnds < block.timestamp || _lockPeriods < block.timestamp) {
+            revert InvalidPeriods();
+        }
+        depositPeriodEnds = _depositPeriodEnds;
+        lockPeriodEnds = _lockPeriodEnds;
+        depositLimit = _depositLimit;
+    }
+
+    function _harvestAndReport() internal view override returns (uint256 _totalAssets) {
+        _totalAssets = totalInvested + asset.balanceOf(address(this));
+    }
+
+    function availableDepositLimit(address /*_owner*/) public view override returns (uint256) {
+        // Return the remaining room.
+        return depositLimit - asset.balanceOf(address(this)) > 0 ? depositLimit - asset.balanceOf(address(this)) : 0;
+    }
+
+    function availableWithdrawLimit(address /*_owner*/) public view override returns (uint256) {
+        return TokenizedStrategy.totalIdle() + totalInvested;
+    }
+
+    function _deployFunds(uint256 _amount) internal override {
+        require(block.timestamp <= depositPeriodEnds, "Deposit period has ended");
+        asset.transfer(TokenizedStrategy.management(), _amount);
+        totalInvested += _amount;
+    }
+
+    function _freeFunds(uint256 _amount) internal override {
+        require(block.timestamp > lockPeriodEnds, "Lock period has not ended");
+        // We don't check available liquidity because we need the tx to
+        // revert if there is not enough liquidity so we don't improperly
+        // pass a loss on to the user withdrawing.
+        asset.safeTransferFrom(TokenizedStrategy.management(), address(this), _amount);
+        totalInvested -= _amount;
+    }
+
+    /**
+     * @dev Optional function for a strategist to override that will
+     * allow management to manually withdraw deployed funds from the
+     * yield source if a strategy is shutdown.
+     *
+     * This should attempt to free `_amount`, noting that `_amount` may
+     * be more than is currently deployed.
+     *
+     * NOTE: This will not realize any profits or losses. A separate
+     * {report} will be needed in order to record any profit/loss. If
+     * a report may need to be called after a shutdown it is important
+     * to check if the strategy is shutdown during {_harvestAndReport}
+     * so that it does not simply re-deploy all funds that had been freed.
+     *
+     * EX:
+     *   if(freeAsset > 0 && !TokenizedStrategy.isShutdown()) {
+     *       depositFunds...
+     *    }
+     *
+     * @param _amount The amount of asset to attempt to free.
+     */
+    function _emergencyWithdraw(uint256 _amount) internal override {
+        uint256 amountToWithdraw = Math.min(_amount, TokenizedStrategy.totalIdle());
+        asset.transfer(TokenizedStrategy.management(), amountToWithdraw);
```

Comment:
totalIdle is a stored value, what if actual "asset.balanceOf(address(this))>totalIdle" looks like we won't be able to withdraw it